### PR TITLE
[FIX] web: fix the unity read with relational properties

### PR DESCRIPTION
**Pull request by std-odoo**
Bug 1
=====
Since 7d2baaa0c726a7b0dd4fe7226862f960c9c59b73 , we read the field with `web_read`. We don't load the display name when calling read, but instead we load them after.

But, since this change, the display names of the relational properties are not loaded.

Bug 2
=====
Since https://github.com/odoo/odoo/commit/8723f020c3587a900c811b8cc23f53fe34b98df3 , they are some tracebacks with the properties when switching the views. For example, if you open a many2many properties record (by clicking on the badge), and then go back to the original form view, a traceback is raised.

Task-3460126

### CODE REVIEW

**Review by thjo-odoo — COMMENTED** [review]
Hello @std-odoo, overall nothing much to say I just have a question linked to the removal of use_display_name that was added with the removal of `name_get`.

Also would it mean that we'd come back to the behavior before the change in 3c62ca1, meaning that we do not need to specify a field dependency if we want the display_name of an invisible many2one ? 

This is  a fix that we did regarding this change: odoo/enterprise#45077. Will it impact this fix ?

File: odoo/fields.py [hunk 1]
```diff
@@ -3333,9 +3334,9 @@ def convert_to_record(self, value, record):
     #       }]
     #
     def convert_to_read(self, value, record, use_display_name=True):
-        return self.convert_to_read_multi([value], record, use_display_name)[0]
```
[the remark this answers is not in the document]
Comment (std-odoo) [reply odoo/fields.py:3336]: my change should not impact other field (and see review message, it's more about "unity read" than the removal of name_get)

**Review by std-odoo — COMMENTED** [review]
@thjo-odoo Hi :) 

> invisible many2one

The thing is that the properties are purely a frontend field, so the display name will be always visible. The bug is not caused by the removal of name_get, but by the removal of the argument "load" in `read` (the argument is still on the method, but it's not used anymore and will be removed at some point).

> Will it impact this fix ?

I don't think, the behavior will be the same as before the "unity read" merge

Cheers :)

**Review by thjo-odoo — APPROVED** [review]

**Review by ryv-odoo — APPROVED** [review]

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**